.github/actions/determine-networks/action.yaml

```diff
@@ -0,0 +1,48 @@
+name: Determine Networks
+description: Detect changed networks from a map of paths.
+inputs:
+  paths_map:
+    required: true
+    description: 'the paths used to check on the network changes'
+
+outputs:
+  networks:
+    value: ${{ steps.set-networks.outputs.networks }}
+    description: Comma-separated list of changed networks or "none"
+
+runs:
+  using: 'composite'
+
+  steps:
+    - name: Get the latest commit SHA
+      shell: bash
+      run: |
+        LATEST_COMMIT=$(git log -1 --pretty=format:"%H")
+        echo "LATEST_COMMIT=$LATEST_COMMIT" >> "$GITHUB_ENV"
+
+    - name: Determine NETWORKS
+      shell: bash
+      id: set-networks
+      run: |
+        echo "Modified files in commit ${{ env.LATEST_COMMIT }}"
+        MODIFIED_FILES=$(git diff --name-only HEAD~1..HEAD)
+        echo "$MODIFIED_FILES"
+
+        NETWORKS=()
+        while IFS='|' read -r NETWORK PATHS_CSV; do
+          IFS=',' read -ra PATHS <<< "$PATHS_CSV"          
+          for PATH in "${PATHS[@]}"; do
+            if [[ "$MODIFIED_FILES" == *"$PATH"* ]]; then
+              echo "Detected change for $NETWORK in $PATH"
+              NETWORKS+=("$NETWORK")
+            fi
+          done
+        done < <(echo '${{ inputs.paths_map }}' | jq -r 'to_entries[] | "\(.key)|\(.value | join(","))"')
+
+        if [ ${#NETWORKS[@]} -eq 0 ]; then
+          echo "No relevant changes detected."
+          echo "networks=none" >> $GITHUB_OUTPUT
+        else
+          echo "Detected networks: ${NETWORKS[*]}"
+          echo "networks=$(IFS=,; echo "${NETWORKS[*]}")" >> $GITHUB_OUTPUT
```

Comment:
unnecessary complexity, let this action deal with a single file/network? The caller workflow can define a matrix to parallelly check multiple files/networks to keep it cleaner

Reply:
@milapsheth coming back to your point, while working on the migration to `gh` CLI, I've came across a limitation about the `gh pr create` that doesn't return a `JSON` response to be able to extract the `PR_NUMBER` for example, so I had to make some changes to be able to get the PR number (with of course avoiding the "hackish" regex parsing of the URL struct and use `gh pr view` instead that returns the PR details in `json` format for example).

Reply:
Unresolved the conversation since the actions are still not refactored to handle a single network at a time

Reply:
The process was indeed "per-network" with the `update-config` action, after the offline sync, I've implemented the "per-network" basis on the whole workflow instead (checking the network's changed files in a specific step instead of a specific action `determine-networks` that was discarded as suggested)